benchopt/plotting/html/static/result.js

```diff
@@ -1,182 +1,307 @@
-    graph = document.getElementById(
-      document.getElementById(nowId).getElementsByTagName("div")[1].id
-    );
-    for (i = 0; i < graph.data.length; i++) {
-      graph.data[i].visible = tracesVisib[i]; // set visible traces for new graph
+const makePlot = () => {
+  const div = document.getElementById('unique_plot');
+  const data = isBarChart() ? getBarData() : getScatterCurves();
+  const layout = isBarChart() ? getBarChartLayout() : getScatterChartLayout();
+
+  Plotly.react(div, data, layout);
+};
+
+const getBarData = () => {
+  return [{
+    type: 'bar',
+    x: useTransformer(data(), null, 'x'),
+    y: useTransformer(data(), null, 'y'),
+  }];
+};
+
+const getScatterCurves = () => {
+  const curves = [];
+
+  getSolvers().forEach(solver => {
+    curves.push({
+      type: 'scatter',
+      name: solver,
+      mode: 'lines+markers',
+      line: {
+        color: data().solvers[solver].color,
+      },
+      marker: {
+        symbol: data().solvers[solver].marker,
+        size: 10,
+      },
+      legendgroup: solver,
+      hovertemplate: solver + ' <br> (%{x:.1e},%{y:.1e}) <extra></extra>',
+      visible: isVisible(solver) ? true : 'legendonly',
+      x: useTransformer(data(), solver, 'x'),
+      y: useTransformer(data(), solver, 'y'),
+    });
+
+    if (state().with_quantiles) {
+      curves.push({
+        type: 'scatter',
+        mode: 'lines',
+        showlegend: false,
+        line: {
+          width: 0,
+          color: data().solvers[solver].color,
+        },
+        legendgroup: solver,
+        hovertemplate: '(%{x:.1e},%{y:.1e}) <extra></extra>',
+        visible: isVisible(solver) ? true : 'legendonly',
+        x: useTransformer(data(), solver, 'q1'),
+        y: useTransformer(data(), solver, 'y'),
+      }, {
+        type: 'scatter',
+        mode: 'lines',
+        showlegend: false,
+        fill: 'tonextx',
+        line: {
+          width: 0,
+          color: data().solvers[solver].color,
+        },
+        legendgroup: solver,
+        hovertemplate: '(%{x:.1e},%{y:.1e}) <extra></extra>',
+        visible: isVisible(solver) ? true : 'legendonly',
+        x: useTransformer(data(), solver, 'q9'),
+        y: useTransformer(data(), solver, 'y'),
+      });
     }
-    Plotly.redraw(graph); // redraw with cohesive visible traces
+  });
+
+  return curves;
+};
+
+/*
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ * Data transformers
+ * 
+ * WARNING : If you add a new transformer function,
+ * don't forget to register it in the object : window.tranformers,
+ * at the end of this section.
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ */
+
+const useTransformer = (data, solver, axis) => {
+  try {
+    let transformer = 'transformer_' + axis + '_' + state().plot_kind;
+    return window.transformers[transformer](data, solver);
+  } catch(e) {
+    console.warn('Trying to call unknown transformer for ' + axis + ' axis. The raw data is returned.');
```

Comment:
Since the function does not return an error if it does not find the transformer and still returns valid data, it will not be obvious that it did not use a transformer (in case of code addition or modification). It's just a reminder that if there's a bug, you might want to look in that direction. I know it's far from optimal, but at least you can't miss it. Should I take it out?

Reply:
`x` is transformed for bar charts.